R/fit.R

```diff
@@ -32,27 +36,46 @@ fit_single_optimizer <- function(formula,
                                  weights,
                                  reml = TRUE,
                                  covariance = NULL,
+                                 tmb_data,
+                                 formula_parts,
                                  ...,
                                  control = mmrm_control(...)) {
-  assert_formula(formula)
-  assert_data_frame(data)
-  assert_vector(weights)
-  assert_flag(reml)
-  assert_class(control, "mmrm_control")
-  assert_list(control$optimizers, names = "unique", types = c("function", "partial"))
-  quiet_fit <- h_record_all_output(
-    fit_mmrm(
-      formula = formula,
-      data = data,
-      weights = weights,
-      reml = reml,
-      covariance = covariance,
-      control = control
-    ),
-    remove = list(
-      warnings = c("NA/NaN function evaluation") # Transient visit to invalid parameters.
+  if (missing(tmb_data) || missing(formula_parts)) {
+    assert_formula(formula)
+    assert_data_frame(data)
+    assert_numeric(weights, any.missing = FALSE, lower = .Machine$double.xmin)
+    assert_flag(reml)
+    assert_class(control, "mmrm_control")
+    assert_list(control$optimizers, names = "unique", types = c("function", "partial"))
+    quiet_fit <- h_record_all_output(
+      fit_mmrm(
```

Comment:
`fit_mmrm` is updated to accept two types of argument, and `fit_mmrm` still works as low level functionality

Reply:
hm but it would be better if we keep named arguments in this `fit_mmrm` call?